### PR TITLE
Docu: users macro logging

### PR DESCRIPTION
**Pull request by teresanunez**

### CODE REVIEW

**Comment by teresanunez** [comment]
The report feature is also added in this PR. 
I was very short in the description since it was already described in the developers part, I put mainly links for not having the same information in different places.

File: doc/source/users/environment_variable_catalog.rst [hunk 1]
```diff
@@ -112,6 +112,19 @@ LogMacroFormat
 
 Definition of the format to be used to save the file.
 
+.. _logmacrofilter:
+
+LogMacroFilter
```
Comment (reszelaz) [inline doc/source/users/environment_variable_catalog.rst]: This is not an environment variable but sardana custom setting. I would put this documentation as a docstring sardanacustomsettings module.

File: doc/source/users/index.rst [hunk 2]
```diff
@@ -16,6 +16,8 @@ User's Guide
     Adding real elements <adding_elements>
     Macro Hooks <macro_hooks>
     Macro Environment <macro_environment>
+    Macro Logging <macro_logging>
```
Comment (reszelaz) [inline doc/source/users/index.rst]: I have no strong opinion about this. But maybe a single chapter describing both features would be enough? At the end they are very similar.

File: doc/source/users/macro_logging.rst [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+.. currentmodule:: sardana.macroserver.macros.logging
+
+
+.. _sardana-macros-logging:
+
+=============
+Macro Logging
+=============
+
+The MacroServer allows to write into a log file messages from the class responsible for
```
Comment (reszelaz) [inline doc/source/users/macro_logging.rst]: This part about "the class responsible running for running the macros" is too specific for the user. Especially as the first sentence describing this feature. I would eventually mention something about additional log entries that enclose each macro execution e.g. when it starts and when it ends. What do you think?

File: doc/source/users/macro_report.rst [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+.. currentmodule:: sardana.macroserver.macros.logging
+
+
+.. _sardana-macros-reports:
+
+=============
+Macro Reports
+=============
+
+:ref:`Report messages <sardana-macro-reporting>` can be send by the macros to a report file. This report feature is configured as described :ref:`here <sardana-configuration-macroserver>`.
```
Comment (reszelaz) [inline doc/source/users/macro_report.rst]: We should [avoid hyperlinks like *here*](http://www.writethedocs.org/guide/writing/docs-principles/#skimmable)

File: doc/source/users/environment_variable_catalog.rst (hunk 1 above)
Comment (reszelaz) [reply doc/source/users/environment_variable_catalog.rst]: You can reference a sardana custom setting with [data Python domain markup](http://www.sphinx-doc.org/en/master/usage/restructuredtext/domains.html#directive-py:data) (see [example](https://sardana-controls.org/users/configuration/spock.html))

Comment (teresanunez) [reply doc/source/users/environment_variable_catalog.rst]: Sorry, it is true ... I implemented and I forgot about it !

**Comment by teresanunez** [comment]
Sorry, I have some problems with this branch, I create a new PR.

**Pull request closed without merging**